quarkchain/cluster/shard.py

```diff
@@ -579,6 +579,36 @@ def broadcast_tx_list(self, tx_list, source_peer=None):
             ]
             consensus_type = shard_config.CONSENSUS_TYPE
             diff = header.difficulty
+
+            # Check difficulty
+            if not self.env.quark_chain_config.SKIP_MINOR_DIFFICULTY_CHECK:
+                if not self.state.db.contain_minor_block_by_hash(
```

Comment:
this requires the prev block to be in DB, but there may be cases where the parent may also still be in the new block pool. This doesn't look right